src/UuidBinaryType.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * This file is part of the Ramsey\Uuid\Doctrine library
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @copyright Copyright (c) Ben Ramsey <http://benramsey.com>
+ * @license http://opensource.org/licenses/MIT MIT
+ * @link https://packagist.org/packages/ramsey/uuid-doctrine Packagist
+ * @link https://github.com/ramsey/uuid-doctrine GitHub
+ */
+
+namespace Ramsey\Uuid\Doctrine;
+
+use InvalidArgumentException;
+use Ramsey\Uuid\Uuid;
+use Doctrine\DBAL\Types\ConversionException;
+use Doctrine\DBAL\Types\Type;
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+
+/**
+ * Field type mapping for the Doctrine Database Abstraction Layer (DBAL).
+ *
+ * UUID fields will be stored as a string in the database and converted back to
+ * the Uuid value object when querying.
+ */
+class UuidBinaryType extends Type
+{
+    /**
+     * @var string
+     */
+    const NAME = 'uuid_binary';
+
+    /**
+     * {@inheritdoc}
+     *
+     * @param array                                     $fieldDeclaration
+     * @param \Doctrine\DBAL\Platforms\AbstractPlatform $platform
+     */
+    public function getSQLDeclaration(array $fieldDeclaration, AbstractPlatform $platform)
+    {
+        return $platform->getBinaryTypeDeclarationSQL(
+            array(
+                'length' => '16',
+                'fixed' => true,
+            )
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @param string|null                               $value
+     * @param \Doctrine\DBAL\Platforms\AbstractPlatform $platform
+     */
+    public function convertToPHPValue($value, AbstractPlatform $platform)
+    {
+        if (empty($value)) {
+            return null;
+        }
+
+        if ($value instanceof Uuid) {
+            return $value;
+        }
+
+        try {
+            $uuid = Uuid::fromBytes($value);
+        } catch (InvalidArgumentException $e) {
+            throw ConversionException::conversionFailed($value, self::NAME);
+        }
+
+        return $uuid;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @param Uuid|null                                 $value
+     * @param \Doctrine\DBAL\Platforms\AbstractPlatform $platform
+     */
+    public function convertToDatabaseValue($value, AbstractPlatform $platform)
+    {
+        if (empty($value)) {
+            return null;
+        }
+
+        if ($value instanceof Uuid || Uuid::isValid($value)) {
+            return hex2bin(str_replace('-', '', (string) $value));
```

Comment:
Hmm, this should in fact be the same. I didn't know the getBytes method existed, so I implemented this here myself.